pkg/ottl/contexts/internal/ctxdatapoint/datapoint.go

```diff
@@ -466,10 +488,12 @@ func accessBucketCounts[K Context]() ottl.StandardGetSetter[K] {
 			return nil, nil
 		},
 		Setter: func(_ context.Context, tCtx K, val any) error {
-			if newBucketCount, ok := val.([]uint64); ok {
-				if histogramDataPoint, ok := tCtx.GetDataPoint().(pmetric.HistogramDataPoint); ok {
-					histogramDataPoint.BucketCounts().FromRaw(newBucketCount)
-				}
+			newBucketCount, err := ctxutil.ExpectType[[]uint64](val)
```

Comment:
We can replace it with `ctxutil.SetCommonTypedSliceValues[uint64]`. The same for other slice paths.